### PR TITLE
Change EveDisplay check for tracks to fatal

### PR DESCRIPTION
**Pull request by DanielWielanek**
Describe your proposal.
1. When FairGeoTracksDraw/FairMCTrackDraw does not find data, it returns kFATAL - which closes the event display. I propose to change it into the kERROR - thanks to this rest of the event display tasks should still work.
2. Another change is using TClass to check if MCTrack branch contains FairMCTrack-based class - if not - then deactivate the task. This prevents the situation when branch "MCTrack" is present but contains a different class (which leads to unexpected behavior).

Mention any issue this PR is resolves or is related to.

---

Checklist:

* [x] Rebased against `dev` branch
* [x] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [ ] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: eventdisplay/tracks/FairEveGeoTracks.cxx
Comment (dennisklein) [inline eventdisplay/tracks/FairEveGeoTracks.cxx:52]:
```suggestion
    if (!fContainer) {
```

Let's use this opportunity and apply [ES.87](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions). Thx!

File: examples/common/eventdisplay/FairEveMCTracks.cxx
Comment (dennisklein) [inline examples/common/eventdisplay/FairEveMCTracks.cxx:152]:
```suggestion
    if (!fContainer) {
```

Same as above, see [ES.87](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions).

File: examples/common/eventdisplay/FairEveMCTracks.cxx
Comment (dennisklein) [inline examples/common/eventdisplay/FairEveMCTracks.cxx:157]:
```suggestion
    if (classTrack && !classTrack->InheritsFrom("FairMCTrack")) {
```

Unless you can make a compelling argument, why this cannot return `nullptr` here?

**Review by dennisklein — CHANGES_REQUESTED** [review]
See my comments. Otherwise, I approve the suggested changes. But better let's wait for one more approval by one of the experts.

File: eventdisplay/tracks/FairEveGeoTracks.cxx
Comment (fuhlig1) [inline eventdisplay/tracks/FairEveGeoTracks.cxx:53]: Could you please add a short explanation what will happen due to the absence of the Geo Tracks. Probably something like "deactivating task".

Please due the same also for the other error messages. It would help the users to understand the consequence.

**Review by fuhlig1 — COMMENTED** [review]
The PR is okay for me after the log messages were updated.

File: examples/common/eventdisplay/FairEveMCTracks.cxx
Comment (DanielWielanek) [reply examples/common/eventdisplay/FairEveMCTracks.cxx]: I decided to use dynamic_cast, I also added this into GeoTracks

**Review by ChristianTackeGSI — COMMENTED** [review]
This looks good to me.

BUT:
* Someone should please verify that the GUI still works (it really should, but nevertheless).
* @DanielWielanek would you mind rebasing / squashing this into a single commit instead of six?

**Review by karabowi — APPROVED** [review]

**Comment by dennisklein** [comment]
https://cdash.gsi.de/testDetails.php?test=12340355&build=347034 is an odd failure, the log output looks just fine. I have seen this already in the past, I suspect it is related to the CMake. However, I think we can exclude that it is cause by this PR at least ;)

**Pull request merged**
